lib/ClangImporter/ClangImporter.cpp

```diff
@@ -993,8 +993,33 @@ ClangImporter::createClangInvocation(ClangImporter *importer,
                                                  &tempDiagClient,
                                                  /*owned*/false);
 
-  return clang::createInvocationFromCommandLine(invocationArgs, tempClangDiags,
-                                                nullptr, false, CC1Args);
+  auto CI = clang::createInvocationFromCommandLine(
+      invocationArgs, tempClangDiags, nullptr, false, CC1Args);
+
+  // FIXME: clang fails to generate a module if there is a `-fmodule-map-file`
+  // argument pointing to a missing file.
+  // Such missing module files occur frequently in SourceKit. If the files are
+  // missing, SourceKit fails to build SwiftShims (which wouldn't have required
+  // the missing module file), thus fails to load the stdlib and hence looses
+  // all semantic functionality.
+  // To work around this issue, drop all `-fmodule-map-file` arguments pointing
+  // to missing files and report the error that clang would throw manually.
+  // rdar://77516546 is tracking that the clang importer should be more
+  // resilient and provide a module even if there were building it.
+  std::vector<std::string> FilteredModuleMapFiles;
+  for (auto ModuleMapFile : CI->getFrontendOpts().ModuleMapFiles) {
+    llvm::sys::fs::file_status status;
+    (void)llvm::sys::fs::status(ModuleMapFile, status);
```

Comment:
Could you look into using the VFS file system instead of checking directly from the file system?
For reference see here where it gets used: https://github.com/apple/swift/blob/main/lib/ClangImporter/ClangImporter.cpp#L1089

Reply:
Updated it to use a VFS, although it’s a bit of a chicken-and-egg problem. We need the compiler invocation to get the VFS and need the VFS to construct the compiler invocation. But I think, it should be fine since the only thing that’s missing from the compiler invocation when we are generating the VFS are the module maps.